sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -0,0 +1,232 @@
+
+    assertThat(result).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.TRACE)).containsExactly(
+      "It seems Deterministic Source Paths are used, replacing '/_/some/path/file.cs' with 'some/path/file.cs'.",
+      "Did not find any indexed file for 'some/path/file.cs'.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_multiple_indexed_files_match_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("root1/some/path/file.cs"), mockInput("root2/some/path/file.cs")));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("/_/some/path/file.cs");
+
+    assertThat(result).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.TRACE)).containsExactly(
+      "It seems Deterministic Source Paths are used, replacing '/_/some/path/file.cs' with 'some/path/file.cs'.");
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Found 2 indexed files for relative path 'some/path/file.cs'. Will skip this coverage entry.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_single_indexed_files_match_returns_file() {
+    InputFile expectedResult = mockInput("root/some/path/file.cs");
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(
+      mockInput("one"),
+      mockInput("two"),
+      expectedResult,
+      mockInput("four")));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("/_/some/path/file.cs");
+
+    assertThat(result).contains(expectedResult);
+    assertThat(logTester.logs(LoggerLevel.TRACE)).hasSize(2);
+    assertThat(logTester.logs(LoggerLevel.TRACE).get(1))
+      .startsWith("Found indexed file ")
+      .endsWith("root/some/path/file.cs' for coverage entry 'some/path/file.cs'.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_with_various_deterministic_source_path_when_match_returns_file() {
+    InputFile expectedResult = mockInput("root/some/path/file.cs");
+    FileSystem fs = createFileSystemForInputFiles(Collections.singletonList(expectedResult));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("C:\\_\\some\\path\\file.cs");
+
+    assertThat(result).contains(expectedResult);
+    assertThat(logTester.logs(LoggerLevel.TRACE)).hasSize(2);
+    assertThat(logTester.logs(LoggerLevel.TRACE).get(0))
+      .isEqualTo("It seems Deterministic Source Paths are used, replacing 'C:\\_\\some\\path\\file.cs' with 'some\\path\\file.cs'.");
+    assertThat(logTester.logs(LoggerLevel.TRACE).get(1))
+      .startsWith("Found indexed file ")
+      .endsWith("root/some/path/file.cs' for coverage entry 'some/path/file.cs'.");
+
+    result = sut.getFilesByRelativePath("D:\\_\\some\\path\\file.cs");
+    assertThat(result).contains(expectedResult);
```

Comment:
`hasValue` calls `contains` directly, but this name seems more readable. The `getFileS` suggests that it's a collection that should contain a member. That's not what the test is doing (even when it goes away after rename)
```suggestion
    assertThat(result).hasValue(expectedResult);
```
Applies below as well.